### PR TITLE
Typescript component conversion

### PR DESCRIPTION
**Pull request by astropunk**
This PR converts many of our Hybrid components to native React Classes, and adds typings. Functionality should be the same.

Xandr employees can see examples/docs [here](https://docspot.adnxs.net/projects/lucid/typescript-component-conversion)

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [ ] Safari
  - [ ] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval

### CODE REVIEW

File: src/components/ExpanderPanel/ExpanderPanel.tsx [hunk 1]
```diff
@@ -16,44 +21,63 @@ const cx = lucidClassNames.bind('&-ExpanderPanel');
...
+
+	/** Indicates that the component is in the "disabled" state when true and in
+			the "enabled" state when false. */
+	isDisabled: boolean;
+
+	/** Controls the presence of padding on the inner content. */
+	hasPadding: boolean;
+
+	/** Called when the user clicks on the component's header.
+			Signature: \`(isExpanded, { event, props }) => {}\` */
```
[the remark this answers is not in the document]
Comment (astropunk) [reply src/components/ExpanderPanel/ExpanderPanel.tsx]: yep I had been trying to, but looks like I missed a couple.  I'll make another pass through.

File: src/components/Expander/Expander.tsx [hunk 2]
```diff
@@ -92,7 +92,7 @@ export interface IExpanderProps
 	/** Renders different variants of Expander. 'simple' is default.
 	 * 'highlighted' is more prominant.
 	 * */
-	kind: 'simple' | 'highlighted';
+	kind?: 'simple' | 'highlighted';
```
[the remark this answers is not in the document]
Comment (astropunk) [reply src/components/Expander/Expander.tsx]: I'll fix it.  I must have missed fixing this one from when I was still wrapping my ahead around how default and required props work together.

File: src/components/TextFieldValidated/TextFieldValidated.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	null;
+TextFieldValidatedError.displayName = 'TextFieldValidated.Error';
+TextFieldValidatedError.peek = {
+	description: `
+		Content that will be displayed as an error message.
+	`,
+};
+TextFieldValidatedError.propName = 'Error';
+
+export interface ITextFieldValidatedProps extends StandardProps {
```
[the remark this answers is not in the document]
Comment (astropunk) [reply src/components/TextFieldValidated/TextFieldValidated.tsx]: yep I think so.  thanks :)

File: src/components/ExpanderPanel/ExpanderPanel.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dsusskind) [reply src/components/ExpanderPanel/ExpanderPanel.tsx]: If we're expecting people will still use the Canvas and look at props in the sidebar then we might want to keep the signatures for now. That sidebar is not using the TS docs and still uses prop types (unless we want to update the sidebar to read from the TS docs, but figured that we want to move to the DocsPage as the main use)

**Pull request merged**